src/formatDate.js

```diff
@@ -7,8 +7,57 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const getSeparator = (format) => {
+    return format.find((part) => /^[^A-Za-z]+$/.test(part));
+  };
+
+  const formatYear = (year, format) => {
+    if (format === 'YYYY') {
+      if (year) {
+        if (year.length === 2) {
+          const prefix = parseInt(year) < 30 ? '20' : '19';
+
+          return prefix + year;
+        } else {
+          return year;
+        }
+      }
+
+      return '';
+    } else if (format === 'YY') {
+      return year ? year.slice(-2) : '';
+    }
+  };
+
+  const parts = date.split(getSeparator(fromFormat));
+
+  const dateParts = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
```

Comment:
The loop condition `i < fromFormat.length - 1` should be `i < fromFormat.length` to ensure all parts of the date are processed. Currently, the last part of the date is not being included in `dateParts`.